### DISCUSSION TITLE
Putting images next to each other

### DISCUSSION THREAD

**Discussion by Sleuth56** [squidfunk/mkdocs-material#3408] [Q&A] [answered]
Is it possible to have two images next to each other horizontally? I've tried to use align left and right with image size and that works on one monitor size but not when that size changes. It tries to fix text under the images in the space instead of just leaving that space blank.

**Comment by squidfunk** [comment, marked as answer]
You don't need to specify alignment if you wish to render images next to each other. Just define them one after another in Markdown, they are rendered as inline elements by default:

``` markdown
![Image 1](...)
![Image 2](...)
```

If you add an empty line after those images, text will start in a new paragraph.

**Reply by samuelcolvin** [reply]
this doesn't seem to work for me, presumably because the images are too large.

**Comment by guzman-raphael** [comment]
Unfortunately this approach did not work for me when trying to center 2 side-by-side images using `figure`

Places them side-by-side :heavy_check_mark: 

```markdown
![Elements](../../images/elements-logo.png){: style="height:83px"}
![Dandi](../../images/dandi-logo.png){: style="height:83px"}
```

Puts them on new lines :x: 

```markdown
<figure markdown>
  ![Elements](../../images/elements-logo.png){: style="height:83px"}
  ![Dandi](../../images/dandi-logo.png){: style="height:83px"}
  <figcaption>Partners</figcaption>
</figure>
```

Got the idea for using [figure](https://squidfunk.github.io/mkdocs-material/reference/images/#image-captions) from the docs but curious if there is a better approach.

@squidfunk Is there anything you'd recommend?

**Reply by facelessuser** [reply]
Material adds some extra styling that forces images within a figure to be rendered as block elements:

```css
.md-typeset figure img {
    display: block;
}
```

I'm not quite sure why Material does this, but if you dislike this behavior, simply override the CSS to display inline.

**Reply by guzman-raphael** [reply]
Thanks @facelessuser! This was the missing puzzle for me and now it works as expected. :smiley:

To assist anyone coming here in the future, the following is the actual config that I used:

Within your markdown use (as before):

```markdown
<figure markdown>
  ![Elements](../../images/elements-logo.png){: style="height:83px"}
  ![Dandi](../../images/dandi-logo.png){: style="height:83px"}
  <figcaption>Partners</figcaption>
</figure>
```

In `mkdocs.yaml`, set the following:

```yaml
docs_dir: ./src
theme:
  name: material
  custom_dir: src/.overrides
```

In `src/.overrides/assets/stylesheets/extra.css`, set the following:

```css
.md-typeset figure img {
    display: inline;
}
```

**Reply by OzGav** [reply]
I'm looking to do the same thing. When I do all of the above the result is the images don't render but they do appear to be centered? Help?

![image](https://github.com/user-attachments/assets/1ce85f8a-b65e-4002-8a88-91a9d56a26fc)

**Reply by kamilkrzyskow** [reply]
Hello @OzGav,
have you properly configured MkDocs? For the `<figure markdown>` to work you need to enable the markdown_extension:
https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#markdown-in-html

If everything is enabled properly, first try to add only 1 image as perhaps some syntax issues is there?
If everything else fails post your raw code here / create a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) with 2 images and all needed files.

**Reply by OzGav** [reply]
OK thanks @kamilkrzyskow that has sort of fixed my problem. I had many extensions but not that one. 

However, whilst the icons are centred they are all one under the other like so:

![image](https://github.com/user-attachments/assets/2f79e9fb-3347-4583-80ff-5dbaca226488)

What I want is for them to be centred and all in a line spilling over and centring as required depending on width of the display so something like:

![image](https://github.com/user-attachments/assets/ddb6467e-41ee-4976-8e1f-fe7506fc61c7)

This is what I currently have

```
<figure markdown>
  ![Apple Music](../assets/icons/apple-music.svg){ width=70 } ![MA](../assets/icon.png){ width=70 } ![Deezer](../assets/icons/deezer-icon.svg){ width=70 } ![Filesystem](../assets/icons/localfiles-icon.png){ width=70 } ![Jellyfin](../assets/icons/jellyfin-logo.svg){ width=70 } ![Preview image](../assets/icons/plex-icon.svg){ width=70 } ![Qobuz](../assets/icons/qobuz-icon.svg){ width=70 } ![Radio Browser](../assets/icons/radiobrowser_icon.png){ width=70 } ![Soundcloud](../assets/icons/soundcloud-icon.svg){ width=70 } ![Spotify](../assets/icons/spotify-icon.svg){ width=70 } ![Subsonic](../assets/icons/subsonic_icon.png){ width=70 } ![Tidal](../assets/icons/tidal-icon.svg){ width=70 } ![TuneIn](../assets/icons/tunein-icon.svg){ width=70 } ![YouTube Music](../assets/icons/ytm-icon.svg){ width=70 }
</figure>
```

**Reply by alexvoss** [reply]
I am not a CSS wizard but this sounds like a use case for [CSS flexboxes](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)?  See [the documentation for adding extra CSS](https://squidfunk.github.io/mkdocs-material/customization/?h=extra+css#additional-css).

**Reply by OzGav** [reply]
Thanks for that. I’m not a CSS wizard either but I’ll see if I can get my head around it.

**Comment by drdv** [comment]
Grid cards seem to provide a nice solution (size of images is adjusted automatically). For example:
``` md
<div class="grid cards" markdown>

-   Daily distance entire challenge

    ---
    ![](summer-walking-challenge/entire_challenge_dist_km.png)

-   Daily distance post-challenge

    ---
    ![](summer-walking-challenge/post_challange_dist_km.png)

-   Daily distance last month of challenge

    ---
    ![](summer-walking-challenge/last_month_dist_km.png)

-   Daily step count last month of challenge

    ---
    ![](summer-walking-challenge/last_month_step_count.png)

</div>
```
This even plays nicely with `glightbox`. Are there potential drawbacks?

**Reply by squidfunk** [reply]
> Are there potential drawbacks?

None that I know of.

---

You can also use images with the [block syntax](https://squidfunk.github.io/mkdocs-material/reference/grids/#block-syntax) (mind the blank lines):

``` html
<div class="grid" markdown>

![Image title](https://dummyimage.com/600x400/f5f5f5/aaaaaa?text=1)

![Image title](https://dummyimage.com/600x400/f5f5f5/aaaaaa?text=2)

![Image title](https://dummyimage.com/600x400/f5f5f5/aaaaaa?text=3)

![Image title](https://dummyimage.com/600x400/f5f5f5/aaaaaa?text=4)

</div>
```

<img width="717" alt="Bildschirm­foto 2024-12-07 um 13 14 40" src="https://github.com/user-attachments/assets/131939b7-7c5c-48c8-93e2-0e7e353f8b49">

The images will stretch to the viewport and switch to 1-column layout on mobile.
